src/shutdown/v1/BaseShutdownStrategy.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Ownable2Step, Ownable} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+
+import {IModuleManager} from "src/interfaces/IModuleManager.sol";
+
+/// @title BaseShutdownStrategy
+/// @author Stake DAO
+/// @dev Provides shared functionality for handling GATEWAY and LOCKER relationships
+abstract contract BaseShutdownStrategy is Ownable2Step {
+    using Math for uint256;
+    using SafeERC20 for IERC20;
+
+    //////////////////////////////////////////////////////
+    /// --- IMMUTABLES
+    //////////////////////////////////////////////////////
+
+    /// @notice The locker contract address
+    address public immutable LOCKER;
+
+    /// @notice The gateway contract address
+    address public immutable GATEWAY;
+
+    /// @notice The default protocol fee.
+    /// @dev The validity of this value is not checked. It must always be valid
+    uint128 internal constant DEFAULT_PROTOCOL_FEE = 0.15e18;
+
+    /// @notice The default harvest fee.
+    /// @dev The validity of this value is not checked. It must always be valid
+    uint128 internal constant DEFAULT_HARVEST_FEE = 0.005e18;
+
+    /// @notice Address of the fee recipient.
+    address public feeRecipient;
+
+    /// @notice The amount of protocol fees accrued.
+    uint256 public protocolFeesAccrued;
+
+    /// @notice Whether the gauge is shutdown
+    mapping(address => bool) public isShutdown;
+
+    /// @notice Error when the gauge is shutdown
+    error SHUTDOWN();
+
+    /// @notice Error when the gauge is not set
+    error ADDRESS_ZERO();
+
+    /// @notice Event when protocol fees are claimed
+    event ProtocolFeesClaimed(uint256 amount);
+
+    //////////////////////////////////////////////////////
+    /// --- CONSTRUCTOR
+    //////////////////////////////////////////////////////
+
+    /// @notice Initializes the base contract with protocol ID, controller, locker, and gateway
+    /// @param _locker The locker contract address (can be zero, in which case GATEWAY is used)
+    /// @param _gateway The gateway contract address
+    /// @param _governance The governance address
+    constructor(address _locker, address _gateway, address _governance) Ownable() {
```

Comment:
The contract is bricked if both the `locker` and the `gateway` are set to 0. Probably worth adding a check for `_gateway`.

```suggestion
    constructor(address _locker, address _gateway, address _governance) Ownable() {
    	if (_gateway_ == address(0)) revert ADDRESS_ZERO();
```